### ISSUE TITLE
org.kohsuke.github.Requester.parse failed to parse GH response

### ISSUE THREAD

**Issue by octo** [collectd/collectd-ci#27]
Error from [pull-requests-github_trigger-aggregation#1815](https://ci.collectd.org/job/pull-requests-github_trigger-aggregation/1815/console):

```
Setting commit status on GitHub for https://github.com/collectd/collectd/commit/7890d5eeedc780e68e4501640bc8a03d4cdcbf12
FATAL: Failed to deserialize {"url":"https://api.github.com/repos/collectd/collectd/statuses/7890d5eeedc780e68e4501640bc8a03d4cdcbf12","id":4297640442,"state":"pending","description":"Build #1815 in progress...","target_url":"https://ci.collectd.org/job/pull-requests-github_trigger-aggregation/1815/","context":"pull-requests-github_trigger-aggregation","created_at":"2017-11-01T12:53:10Z","updated_at":"2017-11-01T12:53:10Z","creator":{"login":"collectd-jenkins","id":12345513,"avatar_url":"https://avatars3.githubusercontent.com/u/12345513?v=4","gravatar_id":"","url":"https://api.github.com/users/collectd-jenkins","html_url":"https://github.com/collectd-jenkins","followers_url":"https://api.github.com/users/collectd-jenkins/followers","following_url":"https://api.github.com/users/collectd-jenkins/following{/other_user}","gists_url":"https://api.github.com/users/collectd-jenkins/gists{/gist_id}","starred_url":"https://api.github.com/users/collectd-jenkins/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/collectd-jenkins/subscriptions","organizations_url":"https://api.github.com/users/collectd-jenkins/orgs","repos_url":"https://api.github.com/users/collectd-jenkins/repos","events_url":"https://api.github.com/users/collectd-jenkins/events{/privacy}","received_events_url":"https://api.github.com/users/collectd-jenkins/received_events","type":"User","site_admin":false}}
java.io.IOException: Failed to deserialize {"url":"https://api.github.com/repos/collectd/collectd/statuses/7890d5eeedc780e68e4501640bc8a03d4cdcbf12","id":4297640442,"state":"pending","description":"Build #1815 in progress...","target_url":"https://ci.collectd.org/job/pull-requests-github_trigger-aggregation/1815/","context":"pull-requests-github_trigger-aggregation","created_at":"2017-11-01T12:53:10Z","updated_at":"2017-11-01T12:53:10Z","creator":{"login":"collectd-jenkins","id":12345513,"avatar_url":"https://avatars3.githubusercontent.com/u/12345513?v=4","gravatar_id":"","url":"https://api.github.com/users/collectd-jenkins","html_url":"https://github.com/collectd-jenkins","followers_url":"https://api.github.com/users/collectd-jenkins/followers","following_url":"https://api.github.com/users/collectd-jenkins/following{/other_user}","gists_url":"https://api.github.com/users/collectd-jenkins/gists{/gist_id}","starred_url":"https://api.github.com/users/collectd-jenkins/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/collectd-jenkins/subscriptions","organizations_url":"https://api.github.com/users/collectd-jenkins/orgs","repos_url":"https://api.github.com/users/collectd-jenkins/repos","events_url":"https://api.github.com/users/collectd-jenkins/events{/privacy}","received_events_url":"https://api.github.com/users/collectd-jenkins/received_events","type":"User","site_admin":false}}
	at org.kohsuke.github.Requester.parse(Requester.java:464)
	at org.kohsuke.github.Requester._to(Requester.java:227)
	at org.kohsuke.github.Requester.to(Requester.java:194)
	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:829)
	at com.cloudbees.jenkins.GitHubSetCommitStatusBuilder.perform(GitHubSetCommitStatusBuilder.java:36)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:534)
	at com.tikal.jenkins.plugins.multijob.MultiJobBuild$MultiJobRunnerImpl.run(MultiJobBuild.java:136)
	at hudson.model.Run.execute(Run.java:1738)
	at com.tikal.jenkins.plugins.multijob.MultiJobBuild.run(MultiJobBuild.java:73)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Numeric value (4297640442) out of range of int
 at [Source: java.io.StringReader@3c24f884; line: 1, column: 122] (through reference chain: org.kohsuke.github.GHCommitStatus["id"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:232)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:197)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1311)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:297)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:121)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:2888)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2034)
	at org.kohsuke.github.Requester.parse(Requester.java:462)
	... 14 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (4297640442) out of range of int
 at [Source: java.io.StringReader@3c24f884; line: 1, column: 122]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1369)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:532)
	at com.fasterxml.jackson.core.base.ParserBase.convertNumberToInt(ParserBase.java:847)
	at com.fasterxml.jackson.core.base.ParserBase.getIntValue(ParserBase.java:643)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseInteger(StdDeserializer.java:306)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:319)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:301)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:464)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:107)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:295)
	... 18 more
```

**Comment by octo** [comment]
> Caused by: com.fasterxml.jackson.databind.JsonMappingException: **Numeric value (4297640442) out of range of int**

That's the value of the `id` field.

**Comment by octo** [comment]
Upstream issue: kohsuke/github-api#387

**Comment by octo** [comment]
Supposedly fixed by github-api-plugin 1.90. (jenkinsci/github-api-plugin#16)

@mfournier Can you upgrade that module / plugin / thingy?

**Comment by mfournier** [comment]
Done, thanks for digging into this @octo !

NB: We still need to wait for the next PR/push to be sure updating this plugin indeed fixes the issue.
